elastic/kibana
# [Remote clusters] Migrate server code out of legacy

This PR completes the NP migration for the `remote_clusters` **server-side** code. I will address the client code in a separate PR.

Changes include:
- Adopted NP router
   - I removed a [workaround in update_route.ts](https://github.com/elastic/kibana/blob/master/x-pack/legacy/plugins/remote_clusters/server/routes/api/update_route.ts#L33) as the issue has been fixed on the ES side.
- Adopted NP license plugin
  - I'm happy to change the implementation here once we decide on a common pattern.
- Updated route tests using the NP core mocks
- Moved server code out of `legacy` directory

_Note: It may be helpful to review the code changes by commit._

### How to test
You will need to set up two clusters in order to test remote clusters. For example:

    yarn es snapshot
    yarn es snapshot -E cluster.name="my_cluster" -E http.port=9201 -E path.data=/tmp/es (in another tab)

- Test all routes and confirm behaving as expected (create, edit, delete, get remote clusters)
- Test license check (remote clusters is available with `Basic` license)
